pkg/utils/validators.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	metricNameValidationRegexStr = `^[a-zA-Z_:]([a-zA-Z0-9_:])*$`
+	labelValidationRegexStr      = `^[a-zA-Z_]([a-zA-Z0-9_])*$`
```

Comment:
Any particular reason we support colon in name but not label?

Reply:
No, this is what Prometheus do. We just mimic same convention.
see: https://github.com/prometheus/client_java/issues/28